pkg/libstack/compose/internal/composeplugin/status_integration_test.go

```diff
@@ -0,0 +1,120 @@
+
+*/
+
+func ensureIntegrationTest(t *testing.T) {
+	if _, ok := os.LookupEnv("INTEGRATION_TEST"); !ok {
+		t.Skip("skip an integration test")
+	}
+}
+
+func TestComposeProjectStatus(t *testing.T) {
+	ensureIntegrationTest(t)
+
+	testCases := []struct {
+		TestName              string
+		ComposeFile           string
+		ExpectedStatus        libstack.Status
+		ExpectedStatusMessage bool
+	}{
+
+		{
+			TestName:       "running",
+			ComposeFile:    "status_test_files/running.yml",
+			ExpectedStatus: libstack.StatusRunning,
+		},
+
+		{
+			TestName:              "failed",
+			ComposeFile:           "status_test_files/failed.yml",
+			ExpectedStatus:        libstack.StatusError,
+			ExpectedStatusMessage: true,
+		},
+	}
+
+	w := setup(t)
+	ctx := context.Background()
+
+	for _, testCase := range testCases {
+		t.Run(testCase.TestName, func(t *testing.T) {
+			projectName := testCase.TestName
+			err := w.Deploy(ctx, []string{testCase.ComposeFile}, libstack.DeployOptions{
+				Options: libstack.Options{
+					ProjectName: projectName,
+				},
+			})
+			if err != nil {
+				t.Fatalf("[test: %s] Failed to deploy compose file: %v", testCase.TestName, err)
+			}
+
+			time.Sleep(5 * time.Second)
+
+			status, statusMessage, err := waitForStatus(w, ctx, projectName, libstack.StatusRunning)
+			if err != nil {
+				t.Fatalf("[test: %s] Failed to get compose project status: %v", testCase.TestName, err)
+			}
+
+			if status != testCase.ExpectedStatus {
+				t.Fatalf("[test: %s] Expected status: %s, got: %s", testCase.TestName, testCase.ExpectedStatus, status)
+			}
+
+			if testCase.ExpectedStatusMessage && statusMessage == "" {
+				t.Fatalf("[test: %s] Expected status message but got empty", testCase.TestName)
+			}
+
+			err = w.Remove(ctx, projectName, nil, libstack.Options{})
+			if err != nil {
+				t.Fatalf("[test: %s] Failed to remove compose project: %v", testCase.TestName, err)
+			}
+
+			time.Sleep(20 * time.Second)
+
+			status, statusMessage, err = waitForStatus(w, ctx, projectName, libstack.StatusRemoved)
+			if err != nil {
+				t.Fatalf("[test: %s] Failed to get compose project status: %v", testCase.TestName, err)
+			}
+
+			if status != libstack.StatusRemoved {
+				t.Fatalf("[test: %s] Expected stack to be removed, got %s", testCase.TestName, status)
+			}
+
+			if statusMessage != "" {
+				t.Fatalf("[test: %s] Expected empty status message: %s, got: %s", "", testCase.TestName, statusMessage)
+			}
+		})
+	}
+}
+
+func waitForStatus(deployer libstack.Deployer, ctx context.Context, stackName string, requiredStatus libstack.Status) (libstack.Status, string, error) {
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	statusCh := deployer.WaitForStatus(ctx, stackName, requiredStatus)
+	select {
+	case result := <-statusCh:
+		if result == "" {
+			return requiredStatus, "", nil
+		}
+
+		return libstack.StatusError, result, nil
+	case <-ctx.Done():
```

Comment:
Same comment as this one https://github.com/portainer/agent/pull/478#discussion_r1254920920